spring-data-cassandra/src/main/java/org/springframework/data/cassandra/convert/ColumnReader.java

```diff
@@ -38,70 +61,66 @@ public Object get(String name) {
 		return get(indexOf);
 	}
 
-	public Object get(int i) {
+	/**
+	 * Read data from a Column using the {@code index}.
+	 *
+	 * @param index
+	 * @return
+	 */
+	public Object get(int index) {
 
-		if (row.isNull(i)) {
+		if (row.isNull(index)) {
 			return null;
 		}
 
-		DataType type = columns.getType(i);
+		DataType type = columns.getType(index);
 
 		if (type.isCollection()) {
+			return getCollection(index, type);
+		}
 
-			List<DataType> collectionTypes = type.getTypeArguments();
-			if (collectionTypes.size() == 2) {
-				return row.getMap(i, collectionTypes.get(0).asJavaClass(), collectionTypes.get(1).asJavaClass());
-			}
+		if (ObjectUtils.nullSafeEquals(type.getName(), Name.TUPLE)) {
```

Comment:
Can also be simplified to...

`if (Name.TUPLE.equals(type.getName()) {`

A properly coded `equals` method handles the `null` case with...

```
if (!(obj instanceof Type)) {
  return false;
}
```